### ISSUE TITLE
[Question] vulnerabilityAlerts for GitLab merge requests?

### ISSUE THREAD

**Issue by ndbroadbent** [renovatebot/config-help#405]
### Which Renovate are you using? CLI, App, or Pro

App

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps

GitLab

### What is your question?

I've set  `"automerge": true` and `"automergeType": "branch"` in my `renovate.json` config, and I just realized that security patches might be silently merged into master, so I probably wouldn't know about it until my next `git pull`. This might mean that there is a delay of a few days before I deploy the updated dependency to production. I also subscribe to some security mailing lists for Ruby/Rails, so hopefully I will get some notifications through different channels. But I'm worried about missing notifications for vulnerabilities in NPM dependencies.

Before Renovate Bot, I had set up a daily CI job that ran `yarn audit` and `bundle audit`, so I would get notified about any security vulnerabilities at least once a day (when the build started crashing.)

Is there any way to ensure that Renovate Bot will always open a Merge Request for vulnerability patches, so that I always see a notification for these?

Thanks very much!

**Comment by rarkins** [comment]
Currently Renovate doesn't have any support for vulnerability alert awareness on GitLab. While GitHub have a formal API to allow apps to query the list of alerts for a repository, GitLab does not. So for now, Renovate is unfortunately unaware if an update it is proposing fixes a vulnerability or not.

If you are interested enough, please raise issues in the main repo (one for JS, one for Ruby) about investigating "audit" integration for both npm and bundler ecosystems. We'd need to do some investigation about how to integrate into those APIs (e.g. via CLI scraping or direct API access) as well as then how to integrate the data into Renovate's logic flow.

**Comment by ndbroadbent** [comment]
Thanks @rarkins! I opened an issue on the main repo. GitLab CI has a [Dependency Scanning](https://docs.gitlab.com/ee/user/application_security/dependency_scanning/) feature, but I just realized that this might not be so helpful, because it only runs during the CI build. So it can only tell us that there are no vulnerabilities in the updated branch, but it doesn't provide any information about security vulnerabilities in the previous version of a package.

So I will open two issues for JS and Ruby.
